Source/UI/NewUI/CommonUI/View/TableViewCell/CKOCellButton.swift

```diff
@@ -0,0 +1,42 @@
+import UIKit
+
+protocol CKOCellButtonDelegate: AnyObject {
+    func buttonIsPressed()
+}
+
+final class CKOCellButton: UITableViewCell {
+    weak var delegate: CKOCellButtonDelegate?
+
+    private var mainView: UIView
+    init(mainView: UIView) {
+        self.mainView = mainView
+        super.init(style: .default, reuseIdentifier: nil)
```

Comment:
This initialiser similarly loses cell dequeuing behaviour. Furthermore, on successive dequeues each cell could end up with unlimited number of stacked views, potentially causing memory leaks and having negative performance impact.

Reply:
Please share with me xcode memory leak screenshot and scenario to fix. 
otherwise 
Please read this 
 https://www.codementor.io/@warrenburton/do-we-need-to-dequeue-tableview-cells-w8szp55uq

Reply:
You have shared this previously and I have been happy to see the comparison, no problem. That is however not describing a commercial project where people come and go, would work perfect on a personal project. 

I simply cannot conceive a case where:
- create a tableView that doesn't dequeue
- create a number of cells that are not cells
- go out of your way to create custom properties and identifiers just to avoid using dequeue behaviour

Is a good idea. Internet is large as we know, I am confident you will find an article to agree with whatever opinion you may want to present, but ultimately you should make the best decision for the use case. 

You build this expecting 8 fields, no problem. 
Next year Gary will come and extend that to 300 fields because that was his requirement. Code review looks legit, local testing passes. Client in prod discovers a bug and we spend a few days to track 1 year old code that's doing _special things_.
3 years on, James, junior dev who hasn't worked commercially before will get his first complex task. He has learned to look around the codebase at similar implementation to get ideas and make sure he has a higher chance to conform to the team's standard. He's never done tableViews before and when comparing a dequeueable implementation over a slap a view over a cell, he finds that slapping a view over a cell is clearer, so he's confident doing this. James will do some more pair programming before he gets another larger individual task and the team has to redo his work.

Save Gary from postmortem and help James learn correctly, stick to the intended use when its a clear cut option of right tool for the right job.

Reply:
> Please share with me xcode memory leak screenshot and scenario to fix.

There isn't one as of yet. However:
- what happens when you stick a view on top of a cell
- that cell is never cleaned by a dequeue
- the cell doesn't get removed from memory, it just lingers on while new cells keep getting created?

Sure, not technically a reference cycle, but a higher than normal memory use nevertheless, when you scroll and cells just keep popping from scratch.

Just because you're not causing a memory leak **now** doesn't mean you're not encouraging a silent one **tomorrow**.